src/components/views/settings/SetIdServer.js

```diff
@@ -0,0 +1,243 @@
+
+    if (parsedUrl.protocol !== 'https:') return _t("Identity Server URL must be HTTPS");
+
+    // XXX: duplicated logic from js-sdk but it's quite tied up in the validation logic in the
+    // js-sdk so probably as easy to duplicate it than to separate it out so we can reuse it
+    return new Promise((resolve) => {
+        request(
+            // also XXX: we don't really know whether to hit /v1 or /v2 for this: we
+            // probably want a /versions endpoint like the C/S API.
+            { method: "GET", url: isUrl + '/_matrix/identity/api/v1' },
+            (err, response, body) => {
+                if (err) {
+                    resolve(_t("Could not connect to ID Server"));
+                } else if (response.status < 200 || response.status >= 300) {
+                    resolve(_t("Not a valid ID Server (status code %(code)s)", {code: response.status}));
+                } else {
+                    resolve(null);
+                }
+            },
+        );
+    });
+}
+
+export default class SetIdServer extends React.Component {
+    constructor() {
+        super();
+
+        let defaultIdServer = abbreviateUrl(MatrixClientPeg.get().getIdentityServerUrl());
+        if (!defaultIdServer) {
+            defaultIdServer = abbreviateUrl(SdkConfig.get()['validated_server_config']['idServer']) || '';
+        }
+
+        this.state = {
+            currentClientIdServer: MatrixClientPeg.get().getIdentityServerUrl(),
+            idServer: defaultIdServer,
+            error: null,
+            busy: false,
+        };
+    }
+
+    _onIdentityServerChanged = (ev) => {
+        const u = ev.target.value;
+
+        this.setState({idServer: u});
+    };
+
+    _getTooltip = () => {
+        if (this.state.busy) {
+            const InlineSpinner = sdk.getComponent('views.elements.InlineSpinner');
+            return <div>
+                <InlineSpinner />
+                { _t("Checking Server") }
+            </div>;
+        } else if (this.state.error) {
+            return this.state.error;
+        } else {
+            return null;
+        }
+    };
+
+    _idServerChangeEnabled = () => {
+        return !!this.state.idServer && !this.state.busy;
+    };
+
+    _saveIdServer = async () => {
+        this.setState({busy: true});
+
+        const fullUrl = unabbreviateUrl(this.state.idServer);
+
+        const errStr = await checkIsUrl(fullUrl);
+
+        let newFormValue = this.state.idServer;
+        if (!errStr) {
+            MatrixClientPeg.get().setIdentityServerUrl(fullUrl);
+            localStorage.removeItem("mx_is_access_token");
+            localStorage.setItem("mx_is_url", fullUrl);
+            newFormValue = '';
+        }
+        this.setState({
+            busy: false,
+            error: errStr,
+            currentClientIdServer: MatrixClientPeg.get().getIdentityServerUrl(),
+            idServer: newFormValue,
+        });
+    };
+
+    _onDisconnectClicked = () => {
```

Comment:
I am not sure that this dialog is really adding that much, since it just confirms removing the server, and we're already making the first button appear dangerous... I wonder if @nadonomy has an opinion.

I think we do want some kind of step that warns about disconnecting the IS when there are 3PIDs currrently bound, as [mentioned in the issue](https://github.com/vector-im/riot-web/issues/10425#issuecomment-517700526).

Reply:
Trying it with no confirmation felt very trigger-happy - I might even argue that for establishing the convention that any danger button has a confirmation.